CS9490/Travel-Itinerary-Project
# Feat fahad

**PR Title:** Add Comprehensive Testing for the Django Project

**PR Description:**
This pull request enhances the project's testing suite by adding various types of tests, including Unit Tests, Integration Tests, End-to-End Tests, and Test Execution Automation. The goal is to improve code quality, detect issues early, and ensure reliable functionality.

**Changes Made:**
- **Unit Tests:** Added unit tests for key components, such as the Itinerary class.
- **Integration Tests:** Introduced integration tests for critical interactions within the application.
- **End-to-End Tests:** Implemented end-to-end tests using Selenium to verify user flows.
- **Test Execution Automation:** Configured GitHub Actions for automated testing on each pull request.

**Testing Steps:**
1. Clone this branch locally.
2. Run the following commands to execute tests:
   - Unit Tests: `python manage.py test`
   - Integration Tests: `python manage.py test travel_itinerary_app.tests.integration_tests`
   - End-to-End Tests: `python manage.py test tests.e2e_tests`
3. Verify that all tests pass without errors.





**Checklist:**
- [ ]x All tests have been added and pass successfully.
- [x ] Code has been reviewed and adheres to coding standards.
- [ x] Documentation has been updated, if necessary.
- [x ] Test coverage has improved.
- [x] PR has been reviewed by a team member.
